### PR TITLE
[OBSDEF-5316] - Keep support store secret unchanged if already exists

### PR DESCRIPTION
**Pull request by fyang86**
## Purpose
[OBSDEF-5316](https://jira.cec.lab.emc.com/browse/OBSDEF-5316)
When we helm upgrade a chart, randAlphaNum inside secret will always change, which caused support store 401 unauthorized error. So we need to keep it unchanged if the secret already exists.

## PR checklist
- [ ] make test passed
- [ ] make build passed
- [ ] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [x] helm upgrade <chart> passed
- [ ] helm test <release_name> passed

## Testing
```
$ kubectl get secret supportassist-objectscale-store-secret -o yaml
apiVersion: v1
data:
  password: WjY3S3RtWDRveQ==
  salt: SWtmaTh5bHVGdw==
  username: b0VtdWU=
```
After helm upgrade:
```
$ kubectl get secret supportassist-objectscale-store-secret -o yaml 
apiVersion: v1
data:
  password: WjY3S3RtWDRveQ==
  salt: SWtmaTh5bHVGdw==
  username: b0VtdWU=
```

### CODE REVIEW

**Comment by eddiepavkovic** [comment]
@fyang86  - but no upgrade was done so how did the secret change?

**Comment by fyang86** [comment]
> @fyang86 - but no upgrade was done so how did the secret change?

@eddiepavkovic I found that the timestamp of support store secret is the same time when helm chart updated.
```
kubectl get secret supportassist-objectscale-store-secret -o yaml
apiVersion: v1
data:
  password: R21lT0g3V2cxYQ==
  salt: MGpGYW5VbHJJZA==
  username: SXc1V0I=
kind: Secret
...
...
    operation: Update
    time: "2021-02-23T00:12:32Z"
```
```
helm list
NAME                     	NAMESPACE                  	REVISION	UPDATED                                  	STATUS  	CHART                     	APP VERSION

supportassist-objectscale	dellemc-eddie-68-domain-c45	1348    	2021-02-23 00:12:31.917143124 +0000 UTC  	failed  	supportassist-2.0.67      	2.0.67     
```
find this on your environment, maybe helm upgrade was evoked by something else.
Anyhow, I think we should keep the secret unchanged if it is using randAlphaNum when we doing helm upgrade.

**Comment by eddiepavkovic** [comment]
@fyang86 - maybe we haven't tested this but are we sure the secret is updated on an upgrade.  i guess it is and thats why we're seeing these issues.

what's weird is that prior to changing the base image of nginx we didn't see this issue.  again maybe because we never tested upgrade as much as we do now.

**Comment by eddiepavkovic** [comment]
helm lint is broken: 
```
==> Linting supportassist
[ERROR] templates/: parse error at (supportassist/templates/supportassist-ese-store-secret.yaml:9): function "lookup" not defined

==> Linting decks-support-store
[ERROR] templates/: parse error at (decks-support-store/templates/support-store-secret.yaml:8): function "lookup" not defined
```

https://asd-ecs-jenkins.isus.emc.com/jenkins/job/monitoring-libs/job/objectscale-charts/job/bugfix-OBSDEF-5316-fix-helm-upgrade-change-randAlphaNum/1/console

**Review by vgupta-mickey — CHANGES_REQUESTED** [review]
Let’s understand the problem here. We can’t reuse the previous secrets all the time, what happens if we really want to change the secrets.
Let’s find out the root cause of this problem.
Every upgrade the secret will change because the templates get run every time.

File: decks-support-store/templates/support-store-secret.yaml [hunk 1]
```diff
@@ -5,7 +5,13 @@ metadata:
   name: decks-support-store-secret
   namespace: {{.Release.Namespace}}
 type: Opaque
```
Comment (vgupta-mickey) [inline decks-support-store/templates/support-store-secret.yaml:7]: add annotation and use pre-intsall and delete-policy
https://github.com/EMCECS/charts/blob/master/ecs-cluster/templates/initial-user-secret.yaml

**Review by eddiepavkovic — CHANGES_REQUESTED** [review]

File: decks-support-store/templates/support-store-secret.yaml [hunk 2]
```diff
@@ -5,7 +5,13 @@ metadata:
   name: decks-support-store-secret
   namespace: {{.Release.Namespace}}
```
[the remark this answers is not in the document]
Comment (fyang86) [reply decks-support-store/templates/support-store-secret.yaml:6]: updated with adding hook, retested and it worked

**Review by eddiepavkovic — APPROVED** [review]
Excellent

**Comment by fyang86** [comment]
@eddiepavkovic @vgupta-mickey there might be another issue using hook, if we `helm delete` support assist chart, this secret won't be deleted until there's new support assist installed. Do we want to keep this secret even if the chart is deleted?

**Review by vgupta-mickey — APPROVED** [review]
it is ok to keep the secrets until the next installment.

**Pull request merged**